src/app/process.go

```diff
@@ -157,12 +164,20 @@ func (p *Process) WaitForCompletion(waitee string) int {
 	return p.procState.ExitCode
 }
 
-func (p *Process) WontRun() {
+func (p *Process) wontRun() {
 	p.onProcessEnd()
 
 }
 
+// perform gracefull process shutdown if defined in configuration
 func (p *Process) shutDown() error {
+	if !p.isState(ProcessStateRunning) {
```

Comment:
This check is messing my use-case up.
With the `pg_ctl start` example, the process goes to `completed` state, since it exits with 0 status code.
In this case the shutdown command not run.
For the sake of simplicity, I think we can ask users to provide an idempotently runnable / non-failing command like:
```
pg_ctl stop || true
```
and run it regardless of process state.